### ISSUE TITLE
Provide a way to use this standalone

### ISSUE THREAD

**Issue by probonopd** [Eldorados/script.module.urlresolver#366]
Is there a way to use this module standalone, i.e., without XBMC?
I would like to use it for https://github.com/probonopd/video2smarttv

**Comment by Eldorados** [comment]
Like all Kodi addons, this one also is tightly tied to it.. without some
very heavy re-work I don't think it would be possible

Have you looked at scripts such as jdownloader?

**Comment by probonopd** [comment]
Thanks for the suggestion, however I am looking for something python-based. It seems to be possible to work around all Kodi-specific stuff and I could imagine this could widen the userbase of this code, especially the plugins, significantly.

**Comment by Eldorados** [comment]
Sorry it's not my interest in doing the work needed, my scope is for Kodi
addons only

There are the 'stubs' available for the required xbmc libraries, you could
try using those and see how far you get -
https://github.com/Tenzer/xbmcstubs

**Comment by J0rdyZ65** [comment]
Actually, the required stubs are very simple: a few tens of python lines.
I built them to test the resolvers/sources without running them from inside the xbmc which is quite tedius.
Have a look at https://github.com/J0rdyZ65/plugin.video.genesit, it is a fork of the old genesis project: the t subdirectory is where my offline driver with xbmc stubs is.

**Comment by probonopd** [comment]
@J0rdyZ65 excellent, but not entirely working yet (unless I am doing something wrong). Although it doesn't error out when importing `urlresolver` it doesn't seem to give the correct result either:

```
>>> import urlresolver
>>> web_url="https://www.youtube.com/watch?v=2a4Uxdy9TQY"
>>> media_url = urlresolver.resolve(web_url)
>>> print media_url
False
```

**Comment by J0rdyZ65** [comment]
Sorry, I should have told you that interactive resolvers (such as openload using the captcha method) cannot be tested offline (unless you implement all the GUI primitives). For a simple driver that *should* work, have a look at my pull on urlresolver which adds the t subdirectory. To use it simply type in the main addon directory: python t URL...
Add -llog to get all the debug output in log file and -i if you have installed mplayer (in /usr/bin) to see the stream metadata such as resolution, audio channels, etc... 
Here is the output I get for a resolvable url:

```
$ pwd
/home/.../GitHub/script.module.urlresolver
$ python t -llog -i http://www.nowvideo.li/video/0e5e71fa0f141
MEDIA URL: http://s201.zerocdn.to/dl/59493ae5221addd3504a98aab08b6c17/56ebb9e6/ff36dc4c89f7cd5a64f35130112b21c862.flv
MEDIA NFO: length: 2:04:14; video: H264 720x400 (274kpbs) ffh264; audio: MP4A 2ch (132kpbs) ffaac
$
```

**Comment by Eldorados** [comment]
There was work already done in this area by lambda and I believe he already
has it running with no modifications needed to urlresolver code

**Comment by J0rdyZ65** [comment]
Indeed, no need to change the urlresolver code: the driver in the pull I submitted uses the addon code as is; it is just a facility for resolvers developers to quickly verify them (at least the resolvers without the need for user interaction).
However, as suggested in the pull, the urlresolver could be enhanced to return also some more meaningful error code (today it just returns False) or at least store it in some class variable so that it can be retrieved later.
Rgds,
G.
